frontend/public/components/cluster-settings/remoteauth-idp-form.tsx

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import * as _ from 'lodash-es';
+
+import { SecretModel, ConfigMapModel } from '../../models';
+import { IdentityProvider, k8sCreate, K8sResourceKind, OAuthKind } from '../../module/k8s';
+import {ButtonBar, PromiseComponent, history, AsyncComponent} from '../utils';
+import { addIDP, getOAuthResource, redirectToOAuthPage } from './';
+import { IDPNameInput } from './idp-name-input';
+import { IDPCAFileInput } from './idp-cafile-input';
+
+const DroppableFileInput = (props: any) => <AsyncComponent loader={() => import('../utils/file-input').then(c => c.DroppableFileInput)} {...props} />;
+
+export class AddRemoteAuthPage extends PromiseComponent<{}, AddRemoteAuthPageState> {
+  readonly state: AddRemoteAuthPageState = {
+    name: 'remoteauth',
+    url: '',
+    caFileContent: '',
+    certFileContent: '',
+    keyFileContent: '',
+    inProgress: false,
+    errorMessage: '',
+  };
+
+  getOAuthResource(): Promise<OAuthKind> {
+    return this.handlePromise(getOAuthResource());
+  }
+
+  createCertSecret(): Promise<K8sResourceKind> {
+    const { certFileContent } = this.state;
+    if (!certFileContent) {
+      return Promise.resolve(null);
+    }
+
+    const secret = {
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        generateName: 'remoteauth-tlsclientcert-secret-',
```

Comment:
Maybe something shorter like

```suggestion
        generateName: 'basic-auth-tls-',
```